packages/flutter/lib/src/material/color_scheme.dart

```diff
@@ -8,163 +8,382 @@ import 'package:flutter/widgets.dart';
 import 'colors.dart';
 import 'theme_data.dart';
 
-/// A set of twelve colors based on the
-/// [Material spec](https://material.io/design/color/the-color-system.html)
+/// A set of colors based on the
+/// [Material spec](https://m3.material.io/styles/color/the-color-system/color-roles)
 /// that can be used to configure the color properties of most components.
 ///
 /// The [Theme] has a color scheme, [ThemeData.colorScheme], which is constructed
 /// with [ColorScheme.fromSwatch].
 @immutable
 class ColorScheme with Diagnosticable {
   /// Create a ColorScheme instance.
+  ///
+  /// For the color parameters that are nullable, it is still recommended
+  /// that applications provide values for them. They are only nullable due
+  /// to backwards compatibility concerns.
+  ///
+  /// If a color is not provided, the closest fallback color from the given
```

Comment:
LGTM. When adding seed color generation, please add a note here to use that instead